extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridBinaryDataConverter.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.EventGrid.TypeConverters
+{
+    /// <summary>
+    /// Converter to bind to BinaryData or BinaryData[] parameter.
+    /// </summary>
+    [SupportedConverterType(typeof(BinaryData))]
+    [SupportedConverterType(typeof(BinaryData[]))]
+    internal class EventGridBinaryDataConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            try
+            {
+                var contextSource = context?.Source as string;
+
+                if (contextSource == null)
+                {
+                    return new(ConversionResult.Failed(new InvalidOperationException("Context source cannot be null")));
```

Comment:
The message here is not quite accurate. `contextSource` would be `null` here if `Source` is not a string, and the message wouldn't be correct.

Something like:
`"The input source should be a non-null string." would be more appropriate.

Extra:  You may also want to add the source type there so we know something came in with a different type if that was ever the case.